### PR TITLE
Add support for f16 tensors

### PR DESCRIPTION
**Pull request by jleibs**
Blocked by: https://github.com/DataEngineeringLabs/arrow2-convert/pull/104

Resolves: https://github.com/rerun-io/rerun/issues/854

```
T = np.random.rand(10, 10, 10).astype(np.float16)
rr.log_tensor("f16_tensor", T)

T = np.random.rand(10, 10).astype(np.float16)
rr.log_image("f16_img_mono", T)

T = np.random.rand(10, 10, 3).astype(np.float16)
rr.log_image("f16_img_rgb", T)
```
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)

### CODE REVIEW

**Comment by emilk** [comment]
Converted to draft since this is blocked on this being merged and released:
* https://github.com/DataEngineeringLabs/arrow2-convert/pull/104

**Comment by jleibs** [comment]
Will need to rebase this once we land: https://github.com/rerun-io/rerun/pull/1455/files

**Review by emilk — APPROVED** [review]
ouch, those are some ugly (but sadly necessary) work-arounds

**Pull request merged**
